.github/workflows/deploy-strapi.yml

```diff
@@ -0,0 +1,42 @@
+name: Build Strapi on Deploy
+
+on:
+   workflow_dispatch:
+   push:
+      branches:
+         - main
+         - deploy-strapi-test*
+      paths:
+         - 'backend/**'
```

Comment:
Since we only rebuild when `backend/**` changes, there won't be a build for every commit. If there's been a frontend pull in the mean time, the `main` branch may point at a commit that we don't have a build for.

Once auto-deploy is setup this likely won't matter as much because we'll be deploy the exact image we just built.